### PR TITLE
[CDAP-14291] Add setup for storybook for CDAP UI

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Adds basic setup for storybook integration with CDAP UI
- Adds some sample storybook for references
- This page is to be used by both UI and design team as a reference for all common set of components that are being used in UI. 
- This is to ensure we have consistency of using components across UI

**Note:**
- This is an ongoing effort. As we introduce more components we could add stories with it to keep this page updated.

JIRA: https://issues.cask.co/browse/CDAP-14291
Build: https://builds.cask.co/browse/CDAP-UDUT96

### CODE REVIEW

File: cdap-ui/.storybook/Welcome.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      alignItems: 'center',
+      background: '#3b78e7',
+      color: 'white',
+    }}>
+      <img src={cdaplogo} />
+      <h1>😍 Welcome to CDAP UI storybook!</h1>
+      <p>
+        Repository of components used in CDAP UI
+      </p>
+      <img src="https://i2.wp.com/blog.codepen.io/wp-content/uploads/2018/07/happy-michael-scott.gif" />
```
Comment (tonybach) [inline cdap-ui/.storybook/Welcome.tsx:40]: lol

File: cdap-ui/package.json [hunk 2]
```diff
@@ -36,19 +38,31 @@
     "@babel/preset-react": "^7.0.0-beta.53",
     "@babel/preset-stage-2": "^7.0.0-beta.53",
     "@babel/preset-typescript": "^7.0.0-beta.53",
+    "@storybook/addon-actions": "^3.4.10",
+    "@storybook/addon-info": "^3.4.10",
+    "@storybook/addon-links": "^3.4.10",
```
Comment (tonybach) [inline cdap-ui/package.json]: I don't see this being used yet. Is it installed by default when you install Storybook?

File: cdap-ui/webpack.config.common.js [hunk 3]
```diff
@@ -154,13 +154,6 @@ if (mode === 'production') {
 var webpackConfig = {
   mode,
   context: __dirname + '/app/common',
-  optimization: {
-    splitChunks: {
-      name: COMMON_LIB_NAME,
-      fileName: COMMON_LIB_NAME + '.js',
-      minChunks: Infinity
-    }
-  },
```
Comment (tonybach) [inline cdap-ui/webpack.config.common.js]: Are these removals related to storybook?

File: cdap-ui/.storybook/config.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+require('font-awesome-sass-loader!../app/cdap/styles/font-awesome.config.js');
+require('../app/cdap/styles/lib-styles.scss');
+require('../app/cdap/styles/common.scss');
+require('../app/cdap/styles/main.scss');
+require('./stories.global.scss');
+require('../app/cdap/styles/bootstrap_4_patch.scss');
+
+// automatically import all files ending in *.stories.js
+const req = require.context('../app/cdap/components/', true, /.stories.tsx$/);
```
Comment (tonybach) [inline cdap-ui/.storybook/config.js:26]: Right now we only have story files ending in `.stories.tsx`, so this is fine, but should we look for `.stories.js` as well?

**Review by tonybach — COMMENTED** [review]
Minor comments and questions.

File: cdap-ui/.storybook/config.js (hunk 4 above)
Comment (ajainarayanan) [reply cdap-ui/.storybook/config.js:26]: Going forward if we write code in `ts` it would be helpful for us to have that static type check. Is there any reason you see why we would want to write stories in `js`?

**Comment by ajainarayanan** [comment]
@elfenheart @tonybach Have added changes to cdap-ui/pom file to build storybook when run from bamboo. 

Once the PR is merged will add another build task for cdap ui develop build that will include building the storybook.

**Review by elfenheart — DISMISSED** [review]
👍 LGTM

**Review by tonybach — APPROVED** [review]
LGTM 👍

**Comment by ajainarayanan** [comment]
Latest green build: https://builds.cask.co/browse/CDAP-UDUT139-2
